### ISSUE TITLE
Concurrent Modification Exception when quickly switching worlds via portal

### ISSUE THREAD

**Issue by Zidane** [SpongePowered/SpongeCommon#761]
```
[22:31:52] [Netty Server IO #8/DEBUG]: Set listener of net.minecraft.network.NetworkManager@2435c246 to net.minecraft.server.network.NetHandlerStatusServer@6508c87d
[22:32:39] [Server thread/FATAL]: Error executing task
java.util.concurrent.ExecutionException: java.util.ConcurrentModificationException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:1.8.0_77]
	at java.util.concurrent.FutureTask.get(FutureTask.java:192) ~[?:1.8.0_77]
	at net.minecraft.util.Util.runTask(Util.java:14) [Util.class:?]
	at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:143) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.updateTimeLightAndEntities(DedicatedServer.java:356) [DedicatedServer.class:?]
	at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:657) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:536) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_77]
Caused by: java.util.ConcurrentModificationException
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1429) ~[?:1.8.0_77]
	at java.util.HashMap$ValueIterator.next(HashMap.java:1458) ~[?:1.8.0_77]
	at net.minecraft.world.gen.structure.MapGenStructure.generateStructure(MapGenStructure.java:85) ~[MapGenStructure.class:?]
	at net.minecraft.world.gen.structure.MapGenStructure.populate(MapGenStructure.java:58) ~[MapGenStructure.class:?]
	at org.spongepowered.common.world.gen.SpongeChunkProvider.populate(SpongeChunkProvider.java:321) ~[SpongeChunkProvider.class:?]
	at net.minecraft.world.gen.ChunkProviderServer.populate(ChunkProviderServer.java:91) ~[ChunkProviderServer.class:?]
	at net.minecraft.world.chunk.Chunk.populateChunk(Chunk.java:1116) ~[Chunk.class:?]
	at net.minecraft.world.gen.ChunkProviderServer.loadChunk(ChunkProviderServer.java:136) ~[ChunkProviderServer.class:?]
	at net.minecraft.server.management.PlayerManager$PlayerInstance.<init>(PlayerManager.java:364) ~[PlayerManager$PlayerInstance.class:?]
	at net.minecraft.server.management.PlayerManager.getPlayerInstance(PlayerManager.java:106) ~[PlayerManager.class:?]
	at net.minecraft.server.management.PlayerManager.updateMountedMovingPlayer(PlayerManager.java:265) ~[PlayerManager.class:?]
	at net.minecraft.server.management.ServerConfigurationManager.serverUpdateMountedMovingPlayer(ServerConfigurationManager.java:324) ~[ServerConfigurationManager.class:?]
	at net.minecraft.network.NetHandlerPlayServer.processPlayer(NetHandlerPlayServer.java:430) ~[NetHandlerPlayServer.class:?]
	at net.minecraft.network.play.client.C03PacketPlayer.processPacket(C03PacketPlayer.java:24) ~[C03PacketPlayer.class:?]
	at net.minecraft.network.play.client.C03PacketPlayer$C04PacketPlayerPosition.processPacket(C03PacketPlayer.java:88) ~[C03PacketPlayer$C04PacketPlayerPosition.class:?]
	at org.spongepowered.common.network.PacketUtil.onProcessPacket(PacketUtil.java:115) ~[PacketUtil.class:?]
	at net.minecraft.network.PacketThreadUtil$1.redirect$onProcessPacket$0(PacketThreadUtil.java:39) ~[PacketThreadUtil$1.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(PacketThreadUtil.java:15) ~[PacketThreadUtil$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_77]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_77]
	at net.minecraft.util.Util.runTask(Util.java:13) ~[Util.class:?]
	... 5 more
[22:32:39] [Server thread/DEBUG]: Failed to add player. EntityPlayerMP['NinjaZidane'/395, l='world', x=2082.03, y=79.39, z=1554.25] already is in chunk 140, 87
[22:32:39] [Server thread/DEBUG]: Failed to add player. EntityPlayerMP['NinjaZidane'/395, l='world', x=2082.03, y=79.39, z=1554.25] already is in chunk 140, 88
[22:32:39] [Server thread/DEBUG]: Failed to add player. EntityPlayerMP['NinjaZidane'/395, l='world', x=2082.03, y=79.39, z=1554.25] already is in chunk 140, 89
[22:32:39] [Server thread/DEBUG]: Failed to add player. EntityPlayerMP['NinjaZidane'/395, l='world', x=2082.03, y=79.39, z=1554.25] already is in chunk 140, 90
[22:32:39] [Server thread/DEBUG]: Failed to add player. EntityPlayerMP['NinjaZidane'/395, l='world', x=2082.03, y=79.39, z=1554.25] already is in chunk 140, 91
[22:32:39] [Server thread/DEBUG]: Failed to add player. EntityPlayerMP['NinjaZidane'/395, l='world', x=2082.03, y=79.39, z=1554.25] already is in chunk 140, 92
```

**Comment by Zidane** [comment]
Probably happened after doing the switching a good 10-15 times in rapid succession.

**Comment by Zidane** [comment]
Closing as I can't reproduce.

**Comment by zediious** [comment]
I've managed to come across a crash very similar to this. I can't be certain exactly what the cause is, but I've noticed that it oftentimes happens after a player teleports to their SkyClaims island, which is oftentimes in a different dimension than they are in. It is very inconsistent, it will go an entire week without happening, then crash three times in quick succession today, only to be steady now. Here is some information, I was not able to identify a specific mod/plugin in here which is why I haven't brought it to SkyClaims, etc. Hoping I  an get a little enlightenment here.

**Server Log:** https://gist.github.com/zediious/7a2873f929cb5f4edd017dae07fc2428

**Server Debug:** https://gist.github.com/zediious/30edc13979784db3514e1aebbe8edaf9

**Crash Report:** https://gist.github.com/zediious/3ba7b27dd32b61a3d704cb363bce781e

The server is running SkyFactory 4 v4.22 w/
**Forge:** 1.12.2-14.23.5.2854
**SpongeForge:** 1.12.2-2838-7.3.0
